wpa_supplicant/ctrl_iface_zephyr.c

```diff
@@ -175,12 +248,32 @@ wpa_supplicant_global_ctrl_iface_init(struct wpa_global *global)
 	priv->global = global;
 	memset(priv->sock_pair, -1, sizeof(priv->sock_pair));
 
-	ret = socketpair(AF_UNIX, SOCK_STREAM, 0, priv->sock_pair);
-	if (ret != 0) {
-		wpa_printf(MSG_ERROR, "socket(PF_INET): %s", strerror(errno));
+	ret = zvfs_eventfd(0, 0);
+	if (ret < 0) {
+		ret = errno;
+		wpa_printf(MSG_ERROR, "eventfd: %s (%d)", strerror(ret), ret);
+		goto fail;
+	}
+
+	priv->sock_pair[0] = ret;
```

Comment:
should we rename this? As we are removing `select NET_SOCKETPAIR`, this might confuse the users, as it's basically two separate sockets.

Reply:
We can certainly rename it, I did not want to do too many changes so it looks still a bit like socketpair even if it is not.